core/server/data/meta/asset_url.js

```diff
@@ -39,11 +39,13 @@ function getAssetUrl(path, isAdmin, minify) {
         // we don't want to concat the path with our favicon url
         output += path;
 
-        if (!config.get('assetHash')) {
-            config.set('assetHash', utils.generateAssetHash());
-        }
+        if (config.get('env') === 'production') {
```

Comment:
It shouldn't have any effect on browser caching - the problem was that we were setting long-expires cache headers in dev mode, we would have had the same problem with or without the hash.

Normally asset hashes would work around the long-expires cache header because they would be re-generated any time the file contents change but we only have a poor-mans asset hash as an escape valve to force a re-download of assets after a server restart - this was why users were reporting they had to restart their server to see any changes. If we didn't have the asset hash they wouldn't see any changes until they cleared their cache.